src/Resolving/NameServer.cs

```diff
@@ -36,18 +35,11 @@ public class NameServer : IResolver
         public bool AnswerAllQuestions { get; set; }
 
         /// <inheritdoc />
-        public async Task<Message> ResolveAsync(
-            Message request,
-            CancellationToken cancel = default(CancellationToken))
+        public Task<Message> ResolveAsync(Message request, CancellationToken cancel = default(CancellationToken))
         {
             var response = request.CreateResponse();
 
-            foreach (var question in request.Questions)
-            {
-                await ResolveAsync(question, response, cancel);
-                if (response.Answers.Count > 0 && !AnswerAllQuestions)
-                    break;
```

Comment:
This test is really required.  The DNS spec states that when multiple questions are present only one question is answered.  The mDNS spec states that all questions are answerex.

Reply:
first question or first found? Im asking because it's very important )

Reply:
I think DNS is dark on this.  Most DNS callers will never send a multi-question request, see https://serverfault.com/questions/742785/multi-query-multiple-dns-record-types-at-once

However, mDNS callers use it to get the all the `A/AAAA` records at the same time.

Reply:
Added back compatibility, if AnswerAllQuestions then Parallel otherwise as it was